include/util/viewport.hpp

```diff
@@ -38,7 +38,10 @@ unsigned getFittedZoom(util::Coordinate south_west, util::Coordinate north_east)
     const double height_ratio = (max_y - min_y) / detail::VIEWPORT_HEIGHT;
     const auto zoom = detail::MAX_ZOOM - std::max(std::log(width_ratio), std::log(height_ratio)) * detail::INV_LOG_2;
 
-    return std::max<unsigned>(detail::MIN_ZOOM, zoom);
+    if (std::isfinite(zoom))
```

Comment:
@TheMarex we're using `std::log` on both `width_ratio` and `height_ratio`. For `log(x)`:

- x < 0 => NaN
- x = 0 => -inf

http://en.cppreference.com/w/cpp/numeric/math/log

Care to elaborate what you want to check instead?